### PR TITLE
Tweak release profile settings for faster code

### LINKED ISSUE #505: Tweak release profile settings
It would be nice to tweak the [release](https://doc.rust-lang.org/cargo/reference/profiles.html#release) profile settings for binary/runtime optimizations. For example, enabling LTO and incrementing the optimization level would make the binary run faster (but makes the compilation slower).

I would like to submit a PR if applicable. Let me know what you think! 🐻 @foresterre

### PR DESCRIPTION
**Pull request by orhun**
closes #505

- enable link time optimization
- reduce the code generation units

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -81,3 +81,10 @@ test_dir = "0.2.0"
 
 [build-dependencies]
 vergen = { version = "7.4.2", default-features = false, features = ["build", "cargo", "git", "rustc"] }
+
+[profile.release]
+opt-level = 3
+debug = false
+panic = "unwind"
```
Comment (foresterre) [inline Cargo.toml:88]: Aren't `opt-level = 3`, `debug = false` and `panic = "unwind"`, defaults for the release profile? (1). As such, I think it is not necessary to re-specify these; what do you think?

(1) https://doc.rust-lang.org/cargo/reference/profiles.html#release
(2) https://doc.rust-lang.org/cargo/reference/profiles.html#profiles

**Review by foresterre — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (orhun) [reply Cargo.toml]: Makes sense! I updated the PR 🐻

**Review by foresterre — APPROVED** [review]

**Comment by foresterre** [comment]
Thanks!

**Pull request merged**